libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerOptions.cs

```diff
@@ -65,6 +65,22 @@ public QnAMakerOptions()
 
         public Metadata[] StrictFilters { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether gets or sets environment of knowledgebase to be called. 
+        /// </summary>
+        /// <value>
+        /// A value indicating whether to call test or prod environment of knowledgebase. 
+        /// </value>
+        public bool IsTest { get; set; }
+
+        /// <summary>
+        /// Gets or sets ranker Types.
+        /// </summary>
+        /// <value>
+        /// Ranker Types.
+        /// </value>
+        public RankerTypes RankerType { get; set; }
```

Comment:
these properties probably need to be exposed in QnAMAkerDialog/QnAMakerDialog2 and .schema files

Reply:
This we can take later as it would also involve schema changes to pick that up on Composer as well. I will add a task item for this.